holidays/countries/british_virgin_islands.py

```diff
@@ -0,0 +1,179 @@
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        dt = self._add_new_years_day(tr("New Year's Day"))
+        if self._year not in {2004, 2010}:
+            self._add_observed(dt, rule=SAT_SUN_TO_NEXT_MON)
+
+        # Lavity Stoutt's Birthday.
+        self._add_observed(
+            self._add_holiday_mar_7(tr("Lavity Stoutt's Birthday")), rule=ALL_TO_PREV_MON
+        )
+
+        # Commonwealth Day.
+        self._add_holiday_2nd_mon_of_mar(tr("Commonwealth Day"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        # Whit Monday.
+        self._add_whit_monday(tr("Whit Monday"))
+
+        # Sovereign's Birthday.
+        name = tr("Sovereign's Birthday")
+        if self._year < 2020:
+            self._add_holiday_2nd_sat_of_jun(name)
+        else:
+            self._add_holiday_2nd_fri_of_jun(name)
+
+        territory_day_dates = {
+            2015: (JUN, 29),
+            2020: (JUN, 29),
+        }
+        name = (
+            # Colony Day.
+            tr("Colony Day")
+            if self._year < 1978
+            # Territory Day.
+            else tr("Territory Day")
+            if self._year < 2021
+            # Virgin Islands Day.
+            else tr("Virgin Islands Day")
+        )
+        if self._year in territory_day_dates:
+            self._add_holiday(name, territory_day_dates[self._year])
+        else:
+            dt = self._add_holiday_jul_1(name)
+            if self._year not in {2006, 2017}:
+                self._add_observed(dt)
+
+        # Emancipation Monday.
+        emancipation_monday = self._add_holiday_1st_mon_of_aug(tr("Emancipation Monday"))
+
+        # Emancipation Tuesday.
+        self._add_holiday(tr("Emancipation Tuesday"), emancipation_monday + timedelta(days=1))
+
+        # Emancipation Wednesday.
+        self._add_holiday(tr("Emancipation Wednesday"), emancipation_monday + timedelta(days=2))
+
+        if self._year < 2021:
+            # Saint Ursula's Day.
+            dt = self._add_holiday_oct_21(tr("Saint Ursula's Day"))
+            if self._year != 2020:
+                self._add_observed(dt)
+
+        if self._year >= 2021:
+            # Heroes and Foreparents Day.
+            self._add_holiday_3rd_mon_of_oct(tr("Heroes and Foreparents Day"))
+
+        if self._year >= 2021:
+            # The Great March of 1949 and Restoration Day.
+            self._add_holiday_4th_mon_of_nov(tr("The Great March of 1949 and Restoration Day"))
+
+        if self._year != 2022:
+            self._add_observed(
+                # Boxing Day.
+                self._add_christmas_day_two(tr("Boxing Day")),
+                rule=SAT_SUN_TO_NEXT_MON_TUE + MON_TO_NEXT_TUE,
+            )
+        else:
+            self._add_christmas_day_two(tr("Boxing Day"))
```

Comment:
```suggestion
        dt = self._add_christmas_day_two(tr("Boxing Day"))
        if self._year != 2022:
            self._add_observed(dt, rule=SAT_SUN_TO_NEXT_MON_TUE + MON_TO_NEXT_TUE)
```
Perhaps Christmas/Boxing Day can be further simplified, I haven't looked at it in detail yet...